src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,66 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
 import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+    private static final Map<ReportStatusCode, ProcedureRequest.ProcedureRequestStatus> STATUS_CODE_MAPPING = Map.of(
+        ReportStatusCode.UNSPECIFIED, ProcedureRequest.ProcedureRequestStatus.UNKNOWN
+    );
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(final Message message) {
+        return message.getServiceReportDetails().getSubject().getClinicalInfo()
+            .map(this::mapPatientClinicalInfo);
+    }
+
+    private ProcedureRequest mapPatientClinicalInfo(final PatientClinicalInfo patientClinicalInfo) {
+        final ProcedureRequest procedureRequest = new ProcedureRequest();
+
+        mapFreeText(patientClinicalInfo, procedureRequest);
+        mapStatus(patientClinicalInfo, procedureRequest);
+        procedureRequest.setIntent(ProcedureRequest.ProcedureRequestIntent.NULL);
+        procedureRequest.setCode(new CodeableConcept().setText("unknown"));
+        procedureRequest.setId(uuidGenerator.generateUUID());
+
+        return procedureRequest;
+    }
+
+    private void mapFreeText(final PatientClinicalInfo patientClinicalInfo, final ProcedureRequest procedureRequest) {
+        final List<Annotation> annotations = patientClinicalInfo.getFreeTexts().stream()
+            .map(FreeTextSegment::getTexts)
+            .map(texts -> String.join(" ", texts))
+            .map(MappingUtils::unescape)
+            .map(text -> new Annotation().setText(text))
+            .collect(Collectors.toList());
+
+        if (annotations.isEmpty()) {
```

Comment:
Doesn't the `FreeTextSegment` parser handle this? The `texts` field is required and if there are no texts then an exception is thrown

Reply:
@n-r-b it doesn't seem it does because if the PatientClinicalInfo would exist but there would be no FreeText the List<FreeTextSegment> would just be empty. As proof, you can look at this unit test [.](https://github.com/nhsconnect/integration-adaptor-lab-results/pull/65/files#diff-a69e2e1a2fdfda10e730d8d87f6c9b214e754c01ab859df3e0449bb1a4c445d5R63-R74) If it would throw the exception, it would do that earlier. However, maybe there are a bit smarter ways to handle this without having this if statement which I wrote :)

Reply:
Oh yes, I see - I was thinking about the `texts` within the `FreeTextSegment` 😄  Thanks for clarifying.